src/WebJobs.Script/Diagnostics/OpenTelemetry/OpenTelemetryConfigurationExtensions.cs

```diff
@@ -0,0 +1,136 @@
+                    {
+                        o.AddAzureMonitorLogExporter(options => options.ConnectionString = azMonConnectionString);
+                    }
+                    o.IncludeFormattedMessage = true;
+                    o.IncludeScopes = false;
+                })
+                // These are messages piped back to the host from the worker - we don't handle these anymore if the worker has OpenTelemetry enabled.
+                // Instead, we expect the user's own code to be logging these where they want them to go.
+                .AddFilter<OpenTelemetryLoggerProvider>("Host.*", _ => !ScriptHost.WorkerOpenTelemetryEnabled)
+                .AddFilter<OpenTelemetryLoggerProvider>("Function.*", _ => !ScriptHost.WorkerOpenTelemetryEnabled)
+                .AddFilter<OpenTelemetryLoggerProvider>("Azure.*", _ => !ScriptHost.WorkerOpenTelemetryEnabled)
+                .AddFilter<OpenTelemetryLoggerProvider>("Microsoft.Azure.WebJobs.*", _ => !ScriptHost.WorkerOpenTelemetryEnabled);
+
+            // Azure SDK instrumentation is experimental.
+            AppContext.SetSwitch("Azure.Experimental.EnableActivitySource", true);
+
+            loggingBuilder.Services.AddOpenTelemetry()
+                .ConfigureResource(r => ConfigureResource(r))
+                .WithTracing(b =>
+                {
+                    b.AddSource("Azure.*");
+                    b.AddAspNetCoreInstrumentation();
+                    b.AddHttpClientInstrumentation(o =>
+                    {
+                        o.FilterHttpRequestMessage = _ =>
+                        {
+                            Activity activity = Activity.Current?.Parent;
+                            return (activity == null || !activity.Source.Name.Equals("Azure.Core.Http")) ? true : false;
+                        };
+                    });
+                    if (enableOtlp)
+                    {
+                        b.AddOtlpExporter();
+                    }
+                    if (!string.IsNullOrEmpty(azMonConnectionString))
+                    {
+                        b.AddAzureMonitorTraceExporter(options => options.ConnectionString = azMonConnectionString);
+                        b.AddLiveMetrics(options => options.ConnectionString = azMonConnectionString);
+                    }
+                    b.AddProcessor(ActivitySanitizingProcessor.Instance);
+                    b.AddProcessor(TraceFilterProcessor.Instance);
+                });
+
+            string eventLogLevel = GetConfigurationValue(EnvironmentSettingNames.OpenTelemetryEventListenerLogLevel, context.Configuration);
+            if (!string.IsNullOrEmpty(eventLogLevel))
+            {
+                if (Enum.TryParse(eventLogLevel, ignoreCase: true, out EventLevel level))
+                {
+                    loggingBuilder.Services.AddSingleton<IHostedService>(serviceProvider => new OpenTelemetryEventListenerService(level));
```

Comment:
If you don't need DI, you can pass the instance directly. Also, use `AddHostedService`

```suggestion
                    loggingBuilder.Services.AddHostedService(new OpenTelemetryEventListenerService(level));
```